### PR TITLE
tracker/neuralnet: Add support for building on Linux.

### PR DESCRIPTION
**Pull request by zhaozhixu**
* Remove tracker-neuralnet cmake code that prevents building on Linux.
* Update onnxruntime calling code for its API changes. onnxruntime has changed GetInputName/GetOutputName to GetInputNameAllocated/GetOutputNameAllocated in API version 12. Add compiling macro tests for that.

Fix issue #1559.

### CODE REVIEW

File: tracker-neuralnet/CMakeLists.txt [hunk 1]
```diff
@@ -1,8 +1,5 @@
 include(opentrack-opencv)
 set(host-spec "${CMAKE_SYSTEM_NAME} ${CMAKE_SYSTEM_PROCESSOR} ${CMAKE_SIZEOF_VOID_P}")
-if(host-spec MATCHES "^Linux i[3-6]86 4$")
```
Comment (sthalik) [inline tracker-neuralnet/CMakeLists.txt:3]: This shouldn't be necessary.

See https://github.com/opentrack/opentrack/blob/master/cmake/FindONNXRuntime.cmake#L36

File: CMakeLists.txt [hunk 2]
```diff
@@ -54,7 +54,7 @@ if(CMAKE_COMPILER_IS_GNUCXX AND NOT __otr_compile_flags_set)
     set(__otr_compile_flags_set TRUE CACHE INTERNAL "" FORCE)
     foreach(i C CXX)
         set(CMAKE_${i}_FLAGS_RELEASE "-O3 -march=native" CACHE STRING "" FORCE)
-        set(CMAKE_${i}_FLAGS "-ggdb -Wall -Wextra -Wpedantic" CACHE STRING "" FORCE)
+        set(CMAKE_${i}_FLAGS "-ggdb -Wall -Wextra -Wpedantic -fopenmp" CACHE STRING "" FORCE)
```
Comment (sthalik) [inline CMakeLists.txt:57]: Don't touch the toplevel makefile.

File: tracker-neuralnet/model_adapters.cpp [hunk 3]
```diff
@@ -310,13 +328,18 @@ std::optional<PoseEstimator::Face> PoseEstimator::run(
 
     try
     {
+        std::vector<const char *> input_names, output_names;
+        std::transform(input_names_.begin(), input_names_.end(), std::back_inserter(input_names),
```
Comment (sthalik) [inline tracker-neuralnet/model_adapters.cpp:332]: Don't allocate memory in the main loop.

**Review by sthalik — CHANGES_REQUESTED** [review]

File: tracker-neuralnet/CMakeLists.txt (hunk 1 above)
Comment (zhaozhixu) [reply tracker-neuralnet/CMakeLists.txt]: Yes, removed in latest patch.

File: CMakeLists.txt (hunk 2 above)
Comment (zhaozhixu) [reply CMakeLists.txt]: OK. Done in cmake of neuralnet.

File: tracker-neuralnet/model_adapters.cpp (hunk 3 above)
Comment (zhaozhixu) [reply tracker-neuralnet/model_adapters.cpp]: Fixed by adding a vector member for C strings.

**Comment by DaWelter** [comment]
Oh I guess I'm a bit late. Regardless. Nice work.
But are you sure you need to copy the names into `std::string`?

**Pull request merged**